dev/run-pip-tests

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+FWDIR="$(cd "`dirname $0`"/..; pwd)"
+cd "$FWDIR"
+
+# Run the tests, we wrap the underlying test script for cleanup and because early exit
+# doesn't always properly exit a virtualenv.
+$FWDIR/dev/run-pip-tests-2
+export success=$?
+
+# Clean up the virtual env enviroment used if we created one.
+if [ -f ./virtual_env_tmp_dir ]; then
```

Comment:
I think that you could combine both this and the `run-pip-tests-2` into a single script if you used Bash exit traps, e.g.

```
function delete_virtualenv() {
  echo "Deleting temp directory $tmpdir"
  rm -rf "$tmpdir"
}
trap delete_virtualenv EXIT
```

and putting that at the top of the script before you actually create the temporary directory / virtualenv.